src/main/java/DisplayPage.java

```diff
@@ -40,19 +52,41 @@ public static void display(String authToken, SlugURLHandler handler){
         //Populate the Combo Box with the Metrics
         metricSelector.setItems(FXCollections.observableArrayList("BurnDown Chart","Cycle Time", "Lead Time"));
 
+        //Input Field for Sprint Input
+        TextField sprintInput = new TextField("Enter Sprint");
+        sprintInput.setPrefWidth(80);
+        sprintInput.setMaxWidth(80);
+
         Button closeBtn=new Button("Submit");
         closeBtn.setOnAction(e->{
             String slugURL= slugInput.getText();
             projectID= Project.getProjectId(authToken,GlobalData.getTaigaURL(),slugURL);
             String selectedOption=metricSelector.getValue();
             System.out.println(projectID);
             System.out.println(selectedOption+" Selected!!");
+
+            //Get First and Last Date of Sprint
+            String sprint = sprintInput.getText();
+            String TAIGA_API_ENDPOINT = "https://api.taiga.io/api/v1";
+            List<JsonNode> sprintDetails = Burndown.getMilestoneStats(authToken, TAIGA_API_ENDPOINT, projectID, sprint);
+
+            String firstDate = sprintDetails.get(1).asText();
+            String lastDate = sprintDetails.get(2).asText();
+
+            if(Objects.equals(selectedOption, "Cycle Time")){
+                try {
+                    CycleTimeGUI ct = new CycleTimeGUI(projectID,authToken,TAIGA_API_ENDPOINT,firstDate,lastDate);
+                    ct.start(new Stage());
+                } catch (Exception ex) {
+                    throw new RuntimeException(ex);
+                }
+            }
         });
         VBox layout=new VBox(10);
-        layout.getChildren().addAll(label,slugInput,metricSelector,closeBtn);
+        layout.getChildren().addAll(label,slugInput,metricSelector,sprintInput,closeBtn);
```

Comment:
can't we ask for sprint details in window for particular metric?

Reply:
Yes we can do that, I will make the change.